### PR TITLE
Fix AttributeEnumWithConverter to work with AdvSeq

### PR DESCRIPTION
**Pull request by olsl21**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nimi-python/blob/master/CONTRIBUTING.md).
~- [ ] I've updated [CHANGELOG.md](https://github.com/ni/nimi-python/blob/master/CHANGELOG.md) if applicable.~
- [x] I've added tests applicable for this pull request

### What does this Pull Request accomplish?

- Make `AttributeEnumWithConverter` inherits from `object` instead of `AttributeEnum`
- Add `_attribute_id` property to `AttributeEnumWithConverter` to be accessed by `create_advanced_sequence` method in NI-DCPower
- Update `fancy_advanced_sequence.py.mako` method template to consider `AttributeEnumWithConverter` a valid attribute class
- Add `test_create_and_delete_advanced_sequence_attribute_enum_with_converter()` system test to NI-DCPower to ensure that attribute that uses enum with converter can be used in an advanced sequence

### What testing has been done?

The newly added system test passed.

### CODE REVIEW

File: build/templates/_attributes.py.mako [hunk 1]
```diff
@@ -122,6 +122,7 @@ class AttributeEnumWithConverter(AttributeEnum):
                 enum value
         '''
         self._underlying_attribute_enum = underlying_attribute_enum
+        self._attribute_id = underlying_attribute_enum._attribute_id
```
Comment (sbethur) [inline build/templates/_attributes.py.mako:125]: Can you make this class and `AttributeEnum` class be derived from `Attribute` class, which has the `_attribute_id` attribute?

**Review by sbethur — CHANGES_REQUESTED** [review]

File: generated/nidcpower/nidcpower/session.py [hunk 2]
```diff
@@ -4880,7 +4880,7 @@ def create_advanced_sequence(self, sequence_name, property_names, set_as_active_
         for prop in property_names:
             if prop not in Session.__base__.__dict__:
                 raise KeyError('{0} is not an property on the nidcpower.Session'.format(prop))
-            if not isinstance(Session.__base__.__dict__[prop], _attributes.Attribute) and not isinstance(Session.__base__.__dict__[prop], _attributes.AttributeEnum):
+            if not isinstance(Session.__base__.__dict__[prop], _attributes.Attribute) and not isinstance(Session.__base__.__dict__[prop], _attributes.AttributeEnum) and not isinstance(Session.__base__.__dict__[prop], _attributes.AttributeEnumWithConverter):
```
Comment (marcoskirsch) [inline generated/nidcpower/nidcpower/session.py:4883]: Shreyas' suggestion I think would get rid of this, which to be honest is a bit of codesmell.

**Review by marcoskirsch — CHANGES_REQUESTED** [review]

File: build/templates/_attributes.py.mako (hunk 1 above)
Comment (olsl21) [reply build/templates/_attributes.py.mako]: Thanks for the suggestion, I've updated it in my latest commit.

File: generated/nidcpower/nidcpower/session.py (hunk 2 above)
Comment (olsl21) [reply generated/nidcpower/nidcpower/session.py]: Yes, I have updated it in my latest commit.

**Review by sbethur — APPROVED** [review]

**Review by marcoskirsch — APPROVED** [review]

**Pull request merged**